x-pack/packages/kbn-elastic-assistant/impl/assistant/use_send_messages/index.tsx

```diff
@@ -29,25 +33,44 @@ interface UseSendMessages {
 }
 
 export const useSendMessages = (): UseSendMessages => {
-  const { assistantStreamingEnabled, knowledgeBase } = useAssistantContext();
+  const { alertsIndexPattern, assistantStreamingEnabled, ragOnAlerts } = useAssistantContext();
   const [isLoading, setIsLoading] = useState(false);
+  const { defaultAllow, defaultAllowReplacement, knowledgeBase } = useSettingsUpdater();
```

Comment:
This is whats requiring the 'refresh' after the settings save. Once the settings are saved you can just fetch the raw data off of the assistantContext ala:

```ts
  const {
    alertsIndexPattern,
    assistantStreamingEnabled,
    defaultAllow,
    defaultAllowReplacement,
    knowledgeBase,
    ragOnAlerts,
  } = useAssistantContext();
```

`useSettingsUpdater` is only needed if you want to update settings and maintain the intermediary state between the update.

Reply:
this works prefectly